hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java

```diff
@@ -653,6 +655,50 @@ public final V get(final long timeout, final TimeUnit unit)
         throw newTimeoutException(timeout, unit);
     }
 
+    // Use when the caller thread is a ForkJoinWorkerThread
+    class ManagedBlocker implements ForkJoinPool.ManagedBlocker {
+        private boolean timed;
+        private long timeoutNanos;
+
+        ManagedBlocker(boolean timed, long timeoutNanos) {
+            this.timed = timed;
+            this.timeoutNanos = timeoutNanos;
+        }
+
+        @Override
+        public boolean isReleasable() {
+            return Thread.currentThread().isInterrupted()
+                    || isDone();
+        }
+
+        @Override
+        public boolean block() throws InterruptedException {
+            if (isReleasable()) {
+                return true;
+            } else if (!timed) {
+                park();
+            } else if (timeoutNanos > 0) {
+                parkNanos(timeoutNanos);
+            }
+            return isReleasable();
+        }
+    }
+
+    private void manageParking(boolean timed, long timeoutNanos) {
```

Comment:
Can we remove the `timed` parameter? Seems like we need to `parkNanos` when `timeoutNanos > 0`, otherwise just `park()`.
also minor typo: `managedParking`?

Reply:
removed `timed` parameter